beard/utils/tests/test_strings.py

```diff
@@ -7,17 +7,38 @@
 # under the terms of the Revised BSD License; see LICENSE file for
 # more details.
 
-"""Test of string helpers.
+"""Tests of string helpers.
 
 .. codeauthor:: Gilles Louppe <g.louppe@cern.ch>
 
 """
 
 from ..strings import asciify
+from ..strings import normalize_personal_name
 
 
 def test_asciify():
+    """Test of asciify"""
     assert asciify("") == ""
     assert asciify("foo") == "foo"
     assert asciify("bèård") == "beard"
     assert asciify("schröder") == "schroder"
+
+
+def test_normalize_personal_name():
+    """Test of normalize_personal_name"""
+    assert normalize_personal_name("Doe, John") == "doe john"
+    assert normalize_personal_name("Doe, J.") == "doe j"
+    assert normalize_personal_name("Doe, J") == "doe j"
+    assert normalize_personal_name("Doe-Foe, Willem") == "doefoe willem"
+    assert normalize_personal_name("Doe-Foe Willem") == "doe foe willem"
```

Comment:
In the first case, "Doe-Foe" is the last name, while in the second case, because the comma is missing, we dont know which word is the last name. 

Then the purpose was to transform last names so that "Doe-Foe" and "DoeFoe" are matched together. This seems to happen often for Asian names, where have either a concatenated name or a name formed by multiple words, separated with dashes. 

At this point, this is merely a heuristic anyway, and we should come up something more robust .